hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/Segment.java

```diff
@@ -179,6 +182,17 @@ public Segment setSize(long size) {
   }
 
   /**
+   * Setting the CellSet of the segment - used only for flat immutable segment for setting
+   * immutable CellSet after its creation in immutable segment constructor
+   * @return this object
+   */
+
+  protected Segment setCellSet(CellSet cellSetOld, CellSet cellSetNew) {
+    this.cellSet.compareAndSet(cellSetOld,cellSetNew);
```

Comment:
Not sure - why should it be a CAS? If this is the same CellSet, the operation is just idempotent, no?